software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/AbstractMongoDBSshDriver.java

```diff
@@ -115,7 +116,12 @@ public void customize() {
     @Override
     public boolean isRunning() {
         try {
-            return MongoDBClientSupport.forServer((AbstractMongoDBServer) entity).ping();
+            if (entity instanceof MongoDBServerImpl && !((MongoDBServerImpl)entity).clientAccessEnabled()) {
```

Comment:
Is there a good reason to single out `MongoDBServerImpl` over the lack of support for replica sets or shards if client access is disabled? I'd prefer that it didn't. There's no reason it can't support the clients, routers and config servers too. I believe that technically the replica set and sharded entities could be written to handle such restrictive environments (by SSHing to machines and executing scripts rather than connecting with `MongoDBClientSupport`).